MapboxCoreNavigation/VisualInstructionFormatter.swift

```diff
@@ -1,58 +0,0 @@
-import Foundation
-import MapboxDirections
-
-/**
- Formatter for creating visual instructions.
- */
-@objc(MBVisualInstructionFormatter)
-public class VisualInstructionFormatter: NSObject {
-    
-    let routeStepFormatter = RouteStepFormatter()
-    
-    /**
-     Creates the optimal text to be displayed given a `RouteLeg` and `RouteStep`.
-     */
-    @objc public func string(leg: RouteLeg?, step: RouteStep?) -> String? {
-        if let currentLeg = leg, let destinationName = currentLeg.destination.name, let step = step, step.maneuverType == .arrive {
-            return destinationName
-        } else if let destinations = step?.destinations {
-            return destinations.joined(separator: NSLocalizedString("DESTINATION_DELIMITER", bundle: .mapboxCoreNavigation, value: " / ", comment: "Delimiter between multiple destinations"))
-        } else if let step = step, step.isNumberedMotorway, let codes = step.codes {
-            return codes.joined(separator: NSLocalizedString("REF_DELIMITER", bundle: .mapboxCoreNavigation, value: " / ", comment: "Delimiter between route numbers in a road concurrency"))
-        } else if let name = step?.names?.first {
-            return name
-        } else if let step = step {
-            return routeStepFormatter.string(for: step)
-        } else {
-            return nil
-        }
-    }
-    
-    /**
-     Creates the optimal instructions to be displayed given a `RouteLeg` and `RouteStep` divided into primary and secondary `Instruction`.
-     */
-    public func instructions(leg: RouteLeg?, step: RouteStep?) -> (Instruction?, Instruction?) {
-        if let currentLeg = leg, let destinationName = currentLeg.destination.name, let step = step, step.maneuverType == .arrive {
-            return (Instruction([.init(NSLocalizedString("ARRIVED_AT_DESTINATION", bundle: .mapboxCoreNavigation, value: "You have arrived", comment: "Instruction when arrived at a destination"))]),
-                    Instruction(destinationName))
```

Comment:
This will regress #419 until the server-side visual instructions account for the developer-supplied waypoint name.

/cc @mapbox/guidance

Reply:
In https://github.com/mapbox/mapbox-navigation-ios/pull/848#issuecomment-348406695, I suggested displaying this same information in the end-of-route UI, which would partially mitigate the regression. (Even if the developer provides a street address, the user wouldn’t see it until they literally arrive at the destination.)

Reply:
@bsudekum, I think we should override the arrival visual instruction to use the developer-provided waypoint name. For now, any developer-provided waypoint name exists only the client side. The client SDK would be temporarily compensating for missing functionality on the server side.

Reply:
Addressed this in https://github.com/mapbox/mapbox-navigation-ios/pull/767/commits/e57be0b1b917187c7f9b7e9ed39adcf48f09929c. If the waypoint is named and it's an arrival step, the secondary instruction is the named waypoint.